skrub/_table_vectorizer.py

```diff
@@ -679,6 +679,16 @@ def _check_X(self, X, reset):
                 "X.toarray() to convert to a dense numpy array."
             )
 
+        # Check Pandas sparse arrays
+        is_sparse_col = [hasattr(X[col], "sparse") for col in X.columns]
+        if any(is_sparse_col):
+            sparse_cols = X.columns[is_sparse_col]
+            raise TypeError(
+                f"Columns {sparse_cols!r} are sparse Pandas series, but dense "
+                "data is required. Use df[col].sparse.to_dense() to convert "
+                "a series from sparse to dense."
+            )
+
         if not isinstance(X, pd.DataFrame):
```

Comment:
I think moving your code after the conversion to pandas would fix the errors you have.